cargo-verify/src/main.rs

```diff
@@ -0,0 +1,677 @@
+        "-Cpanic=abort",
+        "-Warithmetic-overflow", // Detecting errors is good!
+        "-Coverflow-checks=yes",
+        "-Cno-vectorize-loops", // KLEE does not support vector intrinisics
+        "-Cno-vectorize-slp",
+        "-Ctarget-feature=-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-3dnow,-3dnowa,-avx,-avx2",
+        // use clang to link with LTO - to handle calls to C libraries
+        "-Clinker-plugin-lto",
+        "-Clinker=clang-10",
+        "-Clink-arg=-fuse-ld=lld",
+    ]
+    .join(" ");
+
+    if opt.backend != Backend::Seahorn {
+        // Avoid generating SSE instructions
+        rustflags.push_str(" -Copt-level=1");
+    }
+
+    match std::env::var_os("RUSTFLAGS") {
+        Some(env) => {
+            rustflags.push_str(" ");
+            rustflags.push_str(env.to_str().ok_or("not UTF-8")?);
+        }
+        None => (),
+    };
+
+    Ok(vec![
+        (String::from("RUSTFLAGS"), rustflags),
+        (String::from("CRATE_CC_NO_DEFAULTS"), String::from("true")),
+        (String::from("CFLAGS"), String::from("-flto=thin")),
+        (String::from("CC"), String::from("clang-10")),
+    ])
+}
+
+/// Compile a crate for verification.
+/// Return a bcfile for the entire (linked) crate, and c object files that need
+/// to be linked with the bcfile.
+fn compile(opt: &Opt, package: &str, target: &str) -> CVResult<(PathBuf, Vec<PathBuf>)> {
+    let mut cmd = Command::new("cargo");
+    cmd.arg("build").arg("--manifest-path").arg(&opt.cargo_toml);
+
+    if !opt.features.is_empty() {
+        cmd.arg("--features").arg(opt.features.join(","));
+    }
+
+    if opt.tests || !opt.test.is_empty() {
+        cmd.arg("--tests");
+    }
+
+    // The following line is not present because we care about the target It is
+    // there to allow us to use -Clto to build crates whose dependencies invoke
+    // proc_macros.
+    // FIXME: "=="?
+    cmd.arg(format!("--target={}", target))
+        .args(vec!["-v"; opt.verbose])
+        .envs(get_build_envs(&opt)?)
+        .output_info()?;
+    // .env("PATH", ...)
+
+    // Find the target directory
+    // (This may not be inside the crate if using workspaces)
+    info_at!(&opt, 4, "  Getting target directory");
+    let target_dir = get_meta_target_directory(&opt)?;
+
+    // {target_dir}/{target}/debug/deps/{package}*.bc
+    // where the file name has exactly one 1 '.' (because later we add similar
+    // files, with multiple dots, and we don't want them here)
+    // and the file has a main function.
+    let bc_files = glob(
+        &glob::Pattern::escape(
+            target_dir
+                .clone()
+                .append(target)
+                .append("debug")
+                .append("deps")
+                .append(package)
+                .to_str()
+                .ok_or("not UTF-8")?,
+        )
+        .append("*.bc"),
+    )?
+    .filter_map(Result::ok)
+    // Filter only files that have exactly one '.'
```

Comment:
Can you make this comment better?
I think you are doing this to distinguish between files created by Cargo and files created by us (from patch or link) - but I worry that we will forget that really important detail!